llvm/unittests/Analysis/VectorFunctionABITest.cpp

```diff
@@ -140,8 +147,10 @@ TEST_F(VFABIParserTest, OnlyValidNames) {
 }
 
 TEST_F(VFABIParserTest, ParamListParsing) {
-  EXPECT_TRUE(invokeParser("_ZGVnN2vl16Ls32R3l_foo"));
+  EXPECT_TRUE(invokeParser("_ZGVnN2vl16Ls32R3l_foo", "",
```

Comment:
so I think that this test should fail, as the mangled name is for a scalar function called foo, so I think we should faill if the scalar function name does not match.

Reply:
Added vector names to each test, and deriving the scalar one from the mangled name.
There should be tests for each of the names on each function that has a valid parsing.